clib/dpd.js

```diff
@@ -57,9 +56,15 @@
         root += ':' + location.port;
       }
     }
-
+    if (root !== oldRoot) {
+      if (socket && socket.io) {
+        // disconnect socket if we're changing url
+        socket.io.disconnect();
+      }
+      socket = null;
```

Comment:
Yes it should be documented. But for most use cases it shouldn't be a problem because there are no reasons to change the url after connecting once.